launcher/minecraft/Component.cpp

```diff
@@ -103,7 +91,8 @@ std::shared_ptr<class VersionFile> Component::getVersionFile() const
 {
     if (m_metaVersion) {
         if (!m_metaVersion->isLoaded()) {
-            m_metaVersion->load(Net::Mode::Online);
+            // this method is const but the loading of meta changes the information
+            APPLICATION->metadataIndex()->getLoadedVersion(m_metaVersion->uid(), m_metaVersion->version());
```

Comment:
Removing this fixed the segfault

Reply:
this kinda needs to stay. is the issue because the function is mark `const`? could we un-const the function?

Reply:
If I can reproduce it in my PR, let me fix it there.
Also, I don't think is const-related see the sevfault mentioned here: https://github.com/PrismLauncher/PrismLauncher/pull/2547#issuecomment-2188686634

Reply:
yep this is on my side so going to need to fix it in the parent PR

Reply:
I removed the lines from there and it works: the m_metaVersion is set with a nonloaded version only inside ComponentUpdateTask and there it is also resolved so removing that from there should not impact